service/history/historyEngine.go

```diff
@@ -387,11 +387,11 @@ func (e *historyEngineImpl) Start() {
 	e.logger.Info("History engine state changed", tag.LifeCycleStarting)
 	defer e.logger.Info("History engine state changed", tag.LifeCycleStarted)
 
-	e.registerDomainFailoverCallback()
-
 	e.txProcessor.Start()
 	e.timerProcessor.Start()
 
+	e.registerDomainFailoverCallback()
```

Comment:
Add some comment that the callback has to be registered after the task processor starts

Reply:
Done.